docs/src/main/paradox/server/kubernetes.md

```diff
@@ -0,0 +1,71 @@
+# Kubernetes
+
+An Akka gRPC service can be deployed to a Kubernetes cluster just like any other HTTP-based application,
+but if you want to make sure HTTP/2 is used across the board there are some things to consider.
+
+There are 3 ways to use HTTP/2:
+
+* With TLS
+* Without TLS ('h2c'), using the `Upgrade` mechanism to negotiate between using HTTP/1 and HTTP/2
+* Without TLS ('h2c'), without negotiation (which assumes the client has prior knowledge that the server supports HTTP/2)
+
+Kubternetes supports many types of [Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/),
+but [GCE](https://git.k8s.io/ingress-gce/README.md) and [nginx](https://git.k8s.io/ingress-nginx/README.md)
+seem to be the most widely known.
+
+## LoadBalancer Service
+
+By creating a service of type `LoadBalancer` you can expose your services using a direct TCP loadbalancer.
+This means all features supported by the underlying implementation are available.
+
+Limitations of this approach are that you are restricted to 1 service per IP, and
+more advanced features such as TLS termination and path-based routing are not available.
+
+## Nginx Ingress
+
+Nginx, by default, will only support h2c via the `Upgrade` mechanism. Because this mechanism is more
+complicated than the other 2 approaches, client support for this mechanism is not ubiquitous. Also, the
+loadbalancer will use HTTP/1 rather than HTTP/2 to the service. While this may happen to work it is
+not optimal.
+
```

Comment:
```suggestion
Of the three ways to use HTTP/2 presented at the beginning of this page, nginx only supports 
`h2c` with negotiation via the `Upgrade` mechanism. Because the `Upgrade` mechanism is more
complicated than the other 2 approaches (aka "TLS", and "prior knowledge"), client support 
for `Upgrade` mechanism is not ubiquitous. Also, the loadbalancer will use HTTP/1 rather 
than HTTP/2 to the service. While this may happen to work it is not optimal.

```

I tried to remove as many pronouns as possible. But the current writing seems to focus too much in some details and doesn't paint `nginx` as a very valid option. I liked the `LoadBalancer Service` section because it listed the disadvantages very clearly.

--------------------

I'm not sure how to rewrite the last bit. 

> Also, the loadbalancer will use HTTP/1 rather than HTTP/2 to the service. While this may happen to work it is not optimal.